### PR TITLE
Update things to comply with PEP8

### PR DESCRIPTION
**Pull request by ky28059**
@anematode should review this and tell me if anything breaks

### CODE REVIEW

File: scripts/compare_books.py [hunk 1]
```diff
@@ -140,18 +151,21 @@ def problem_tokenizer(file_string):
...
             yield Token(token_type, match_str)
 
-# Must consider: \begin{enumerate}, \end{enumerate}, \item, \setcounter{...}{...}, \begin{outer_problem}, \begin{inner_problem}, \begin{iinner_problem}
+
+# Must consider: \begin{enumerate}, \end{enumerate}, \item, \setcounter{...}{...}, \begin{outer_problem},
+# \begin{inner_problem}, \begin{iinner_problem}
 def problem_generator(file_string, is_answer_key=False):
     global counters
 
+    # no idea what's supposed to be local and what's supposed to shadow, so am just leaving it like this
```
Comment (ky28059) [inline scripts/compare_books.py:160]: @anematode

**Comment by ytchang05** [comment]
Will close temporarily and reopen in hopes that de-bugged PR workflow will trigger this time.

**Comment by ytchang05** [comment]
Workflow, please run...

**Comment by ky28059** [comment]
Some other recommendations I will make:

- gen_answer_key.py is not used so you should delete (its archived in git history already and keeping it makes the repo bigger but also makes pycharm freak out about duplicated code)

- unfinished code should probably not be committed (generate_matrix_mult.py trails off after an if, the comment in build.py abruptly stops)

- less name shadowing so that its more apparent whats supposed to be a declaration and whats an assignment which has mistakenly become a declaration because of python 3

File: scripts/compare_books.py (hunk 1 above)
Comment (ky28059) [reply scripts/compare_books.py:160]: I am assuming that the one in the beginning is a declaration and the ones later in the if are an assignment?
```py
def problem_generator(file_string, is_answer_key=False):
    global counters

    enum_depth = 0  # declaration?
    record_problem_n = 0

    set_counter_by_value("outer", 0)
    set_counter_by_value("inner", 0)
    set_counter_by_value("iinner", 0)

    for x in range(1, 4):
        set_counter_by_value("enum" + "i" * x, 0)

    for token in problem_tokenizer(file_string):
        if token.type == "begin_enumerate":
            name = token.get_enumeration_name()

            if name == "enumerate":
                enum_depth += 1  # assignment?

                set_counter_by_value("enum" + "i" * enum_depth, 0)
            else:
                if name == "outer_problem":
                    enum_depth = 1  # assignment?
```
If the above assumption is true, the assignments in the else of the if are not properly reassigning enum_depth to their values.

**Comment by ytchang05** [comment]
What that workflow *meant* to say is:
> The edits were built successfully! View the gatm-textbook artifact from this workflow [here](https://github.com/Gunn-GAtM/gatm/actions/runs/576767197).

Turns out that a workflow run on a PR by someone w/o repo perms doesn't have comment perms either, so I will have to find a way to fix that in the future, however the built textbook artifact *did* upload.

File: scripts/compare_books.py (hunk 1 above)
Comment (ky28059) [reply scripts/compare_books.py:160]: Screenshot showing that the later ones are assignments
screenshot

**Review by ytchang05 — COMMENTED** [review]
Everything here looks ok so far, according to the workflow artifact the build still functions correctly, will take a closer look soon at the extra suggestions before merging.

**Comment by anematode** [comment]
I'll be honest, I don't remember how `compare_books.py` works at all. But I'll take a look at some point!

**Comment by ky28059** [comment]
Merge conflicts are painful, please stop committing to main before reviewing and merging this PR @anematode!

**Comment by ky28059** [comment]
Closing and reforking because new commits

**Pull request closed without merging**